### ISSUE TITLE
Comment Node Issues

### ISSUE THREAD

**Issue by shellscape** [webschik/postcss-less#21]
Related to #20, but separate. There are a few issues with the `Comment` Node. Consider the following input: `// Hello world`

Results in the following `Comment` Node:

```js
Comment {
  raws: { before: '', inline: true, left: ' ', right: '' },
  type: 'comment',
  parent: 
   Root {
     raws: { semicolon: false, after: '' },
     type: 'root',
     nodes: [ [Circular] ],
     source: { input: [Object], start: [Object] },
     lastEach: 1,
     indexes: {} },
  source: 
   { start: { line: 1, column: 1 },
     input: Input { css: '// Hello world', id: '<input css 2>' },
     end: { line: 1, column: 14 } },
  text: 'Hello world' }
```

With the following result for `node.toString()`: `/* Hello world*/`

1. It would be reasonable to assume that the preceding characters `//` be somewhere in the `raws` property.
2. The `.toString()` output looks a little off. Should the `.toString()` method assume a space between the `node.text` of 'Hello world' and the ending `*/` for single line comments that are transformed to the CSS standard?
3. The original form of the comment (`// Hello world`) isn't present anywhere in the node's properties, and we can't use `node.source.input.css` since it'll contain all css passed in the current operation. This is an important issue for inspection of comments.

**Comment by webschik** [comment]
@shellscape,

1. According to [PostCSS API](https://github.com/postcss/postcss/blob/master/docs/api.md#noderaws) there is no special property for that. I'm not sure that we really need '//' in raws and how exactly name this special property.

2. It's a [standard output](https://github.com/postcss/postcss/blob/master/lib/node.es6#L61) for `.toString()` method in PostCSS API. If you need specific stringifier for LESS you may use the next code:

    ````js
    import postCssLess from 'postcss-less';
    
    const root = postCssLess.parse('// Hello world');
    const less = root.first.toString({
        stringify: postCssLess.stringify
    }); // it will be '// Hello world'

    ````

3. It's close to point 1. Probably I should add some raw content of comment to the `node.raws`;

By the way `postcss-scss` has the same results and behavior :)

**Comment by shellscape** [comment]
1. There are two functions to the `raws` property: to contain properties for space around a thing, and to contain properties for other metadata that belongs to a thing. There is already a pattern set for that, so I must disagree.

2. I don't know what you're trying to say here. There is no 'standard output' for `toString`, only the default functions found in the `super` class, which can be overriden, and frequently is. By assuming that the user wants comments represented a certain way by `toString` is wrong, but postcss-less (and yes, even postcss-scss) are making the assumption that there should be no space between the last character of the comment text and the closing `*/` which is contrary to the most popular style of writing comments (as evidenced by the default settings of nearly every linter out there which looks at comment style). I don't understand the pushback on this one.

**Comment by webschik** [comment]
2. How it works now:
    Common `PostCSS` stringifier: '// Hello world' => '/* Hello world*/'
    `postcss-less` stringifier: '// Hello world' => '// Hello world'
 
    So, you propose to patch default stringifier and add space before '*/'. In this case it's an issue for       `PostCSS`, probably its API should support most popular style of block comments.

**Comment by shellscape** [comment]
Inline comments are not valid CSS, they're specific to SCSS and LESS (and a few other custom preprocessors). The issue is here: https://github.com/webschik/postcss-less/blob/aaacb1b6cd109c0604a4cf130c65748c6184b1b9/lib/less-parser.js#L13 and how inline comments are made PostCSS `Comment` nodes. If you inspect a typical `/* ... */` comment in PostCSS, you'll see that it adds a space to `raws.after` which the stringifier takes into account. Because you're dropping the extra space and not telling the `Comment` node to include it, it won't have one. 

This is an issue with postcss-less, not PostCSS. IMHO the space should be added by default with an option to disable.

**Comment by webschik** [comment]
@shellscape, I see, I'm sorry :) 
This part of code was added not by me, so it's a little bit under control now. I'm going to fix this issue asap. Thank you.

**Comment by webschik** [comment]
@shellscape, I've fixed it, 057638ff3bd6eb79647da094164b189994108152.

 Please, check [the docs](https://github.com/webschik/postcss-less/blob/master/README.md#comment-node).
